### PR TITLE
dev-embedded/esptool: version bump to 3.3.1, 4.1, remove old versions

### PR DESCRIPTION
**Pull request by madmartin**
dev-embedded/esptool: version bump to 3.3.1, 4.1
    
esptool-3.3.1: last 3.x release
esptool-4.1: new 4.x release with breaking changes

dev-embedded/esptool: remove obsolete unstable versions
remove esptool-3.2-r1
remove esptool-3.3

Package-Manager: Portage-3.0.30, Repoman-3.0.3
Signed-off-by: Martin Dummer <martin.dummer@gmx.net>

### CODE REVIEW

**Comment by thesamesam** [comment]
1. What breaking changes? Describe them in the commit message (why not?)
2. Please add both new versions in separate commits.
3. Please add a blank line after your commit message and before the trailers, e.g.
```
app-misc/foo: add 1.0.0

1.0.0 is a release with breaking changes like ...

Bug: https://bugs.gentoo.org/123456
Signed-off-by: ...
```

**Comment by madmartin** [comment]
> 1.  2. 3. 
@thesamesam ping
adjusted as requested

**Comment by madmartin** [comment]
Please dont merge yet, just a minute ago version 4.0.1 arrived. Will adjust this PR.

**Comment by madmartin** [comment]
@thesamesam  Should be ready to merge now.

**Review by juippis — APPROVED** [review]
Everything seems to work. Do we want to add some sort of `ewarn` based on `REPLACING_VERSIONS` to 4.0.1?

**Comment by madmartin** [comment]
> Everything seems to work. Do we want to add some sort of `ewarn` based on `REPLACING_VERSIONS` to 4.0.1?

Ah, yes, quite good idea!
BTW.... there is version 4.1 out now... will adjust again.

**Comment by madmartin** [comment]
> Everything seems to work. Do we want to add some sort of `ewarn` based on `REPLACING_VERSIONS` to 4.0.1?

@juippis is implemented now. You may take a look.

File: dev-embedded/esptool/esptool-4.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"${EPYTHON}" test/test_imagegen.py || die "test_imagegen.py failed with ${EPYTHON}"
+	"${EPYTHON}" test/test_espsecure.py || die "test_espsecure.py failed with ${EPYTHON}"
+	"${EPYTHON}" test/test_espefuse_host.py || die "test_espefuse_host.py failed with ${EPYTHON}"
+	"${EPYTHON}" test/test_merge_bin.py || die "test_merge_bin.py failed with ${EPYTHON}"
+	"${EPYTHON}" test/test_modules.py || die "test_modules.py failed with ${EPYTHON}"
+	# test/test_esptool.py and test/test_espefuse.py need real hardware connected
+}
+
+pkg_postinst() {
+	if ver_test "${REPLACING_VERSIONS}" -lt "4"; then
```
Comment (juippis) [inline dev-embedded/esptool/esptool-4.1.ebuild:50]: This breaks, I believe "4" will treat is as a string and not an integer. 
```suggestion
	if ver_test "${REPLACING_VERSIONS}" -lt 4; then
```
works, I'll fix while merging. Thank you!

**Review by juippis — APPROVED** [review]
```
ModuleNotFoundError: No module named '_cffi_backend'
FAILED (errors=18)
 * ERROR: dev-embedded/esptool-4.1::gentoo failed (test phase):
 *   test_espsecure.py failed with python3.10
```
You need dev-python/cffi as a test dep. I'll add it!

File: dev-embedded/esptool/esptool-4.1.ebuild (hunk 1 above)
Comment (madmartin) [reply dev-embedded/esptool/esptool-4.1.ebuild:50]: Hello @juippis ,
its okay to change, but I tested my version and it had worked (displaying the warnings when upgrading from 3.3.1 to 4.0.1)

**Pull request closed without merging**